paasta_tools/marathon_tools.py

```diff
@@ -223,14 +223,15 @@ def get_instances(self):
         else:
             return 0
 
-    def get_backoff_seconds(self):
+    def get_backoff_seconds(self, instances=None):
         """backoff_seconds represents how many seconds a penalization factor for failing tasks.
         Every time a task fails, Marathon adds this value multiplied by a backoff_factor.
         In PaaSTA we know how many instances a service has, so we adjust the backoff_seconds
         to account for this, which prevents services with large number of instances from
         being penalized more than services with small instance counts. (for example, a service
         with 30 instances will get backed off 10 times faster than a service with 3 instances)."""
-        instances = self.get_instances()
+        if instances is None:
```

Comment:
@solarkennedy calling `format_marathon_app_dict` calls both `get_instances` and `get_backoff_seconds`. Since `instances` might read data from zookeeper this allows us to only query zookeeper